src/nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -1,36 +1,65 @@
-"""Authors: Cody Baker and Ben Dichter."""
+"""Authors: Cody Baker, Heberto Mayorquin and Ben Dichter."""
 from datetime import datetime
 from pathlib import Path
 from typing import Optional
 
-from spikeextractors import SpikeGLXRecordingExtractor, SubRecordingExtractor, RecordingExtractor
+import spikeextractors as se
+from spikeinterface import BaseRecording
+from spikeinterface.extractors import SpikeGLXRecordingExtractor
+from spikeinterface.core.old_api_utils import OldToNewRecording
+
+from spikeextractors import SubRecordingExtractor, RecordingExtractor
 from pynwb.ecephys import ElectricalSeries
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ..baselfpextractorinterface import BaseLFPExtractorInterface
-from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType
-
-
-def fetch_spikeglx_metadata(file_path: FilePathType, recording: RecordingExtractor, metadata: dict):
-    file_path = Path(file_path)
-    session_id = file_path.parent.stem
-
-    if isinstance(recording, SubRecordingExtractor):
-        n_shanks = int(recording._parent_recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._parent_recording._meta["fileCreateTime"]).astimezone()
+from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType, dict_deep_update
+
+
+def fetch_spikeglx_metadata(recording: BaseRecording, metadata: dict):
+    """
+    Fetches the session_start_time from the meta readings and adds it to the metadata.
+    """
+
+    # Support for old spikeextractors objects
+    recording = recording._kwargs.get("oldapi_recording_extractor", recording)
+    if isinstance(recording, RecordingExtractor):
+        if isinstance(recording, SubRecordingExtractor):
+            meta = recording._parent_recording._meta
+        else:
+            meta = recording._meta
+        # imDatPrb_type 0 and 21 correspond to single shank channels
+        # see https://billkarsh.github.io/SpikeGLX/help/imroTables/
+        imDatPrb_type = meta["imDatPrb_type"]
+        if imDatPrb_type not in ["0", "21"]:
+            raise NotImplementedError(
+                "More than a single shank is not supported in spikeextractors, use the new spikeinterface."
+            )
```

Comment:
So as I understand the slack discussion, this sort of thing would actually more properly be done in the future by using ProbeInterface?